services/editor/src/pages/context/components/FixedKeys/FixedKeys.js

```diff
@@ -26,35 +28,54 @@ const extractLocal = R.pipe(
   R.pluck('local'),
 );
 
-const FixedKeys = ({ className, appendKey, onChange, toggleDelete, keys }) => (
-  <div className={classnames('fixed-keys-container', className)} data-comp="fixed-keys">
-    <div className="override-keys-title">Override Keys</div>
+export default class FixedKeys extends React.Component {
+  static getDerivedStateFromProps({ local, remote }) {
+    const formattedKeys = extractKeys(remote, local);
+    return { formattedKeys };
+  }
 
-    <FixedKeysList {...{ keys, onChange, toggleDelete }} />
+  state = { formattedKeys: undefined };
```

Comment:
What's the use of state here? (it doesn't look like it's updated)

Reply:
it's updated in getDerivedStateFromProps

Reply:
there won't be any use for at when we move to hooks

Reply:
Couldn't it be just getter?

Reply:
I didn't want to calculate it every time.
I started upgrading to hooks (🎉) so it will go away